File: Projects/Project-Euler/P26.js
m => {
  let
  maxIndex = 0,
  maxLength = 0;

  for (let i = m; i > 1; --i) {
  
    let
    counter = 0,
    dividend = 1,
    length = 0,
    reciprocals = [],
    remainder = 1;
    
    while (true) {
      dividend = remainder * 10;
      remainder = dividend % i;
      
      if (remainder === 0) {
        break;
      }
      if (reciprocals[remainder]) {
        length = counter - reciprocals[remainder];
        break;
      }
      
      reciprocals[remainder] = counter++;
    }
    
    if (length > maxLength) {
      maxIndex = i;
      maxLength = length;
    }
  }
  return maxIndex;
}

File: Projects/Project-Euler/P26.py
max = 0
result = 0

for denominator in range(2, 1000):
  counter = 0
  remainder = 1
  remainders = []

  while remainder:
    while remainder < denominator:
      counter += 1
      remainder = remainder * 10

    remainder %= denominator

    if remainder in remainders:
      break

    remainders.append(remainder)

  if max < counter:
    max = counter
    result = denominator

print(result)